### DISCUSSION TITLE
Upload File with Safe characters in Key causes Forbidden Error

### DISCUSSION THREAD

**Discussion by Playmountain710** [aws/aws-sdk-net#2604] [General]
Hi, 
i am trying to Upload files in our bucket. The file Key is looking something like that: 
`"file_(test)_file.csv"`

But the Upload is failing with Error "Fobidden" - "The request signature we calculated does not match the signature you provided. Check your key and signing method"

Here is my code for the upload: 

    
```
 _logger.LogInformation($"Write {obj.Name} to S3 Bucket {settings.ObsBucketName}");
            var credentials = new BasicAWSCredentials(settings.ObsAccessKey, settings.ObsSecretKey);
            var config = new AmazonS3Config()
            {
                ServiceURL = settings.ObsAccessPoint,
                ForcePathStyle = true
            };
            var client = new AmazonS3Client(credentials, config);
            var response = new S3ResponseDto();
```

            try
            {
                var uploadRequest = new TransferUtilityUploadRequest()
                {
                    InputStream = obj.InputStream,
                    Key = obj.Name,
                    BucketName = obj.BucketName,
                    CannedACL = S3CannedACL.NoACL,
                    //DisableMD5Stream = true,
                    //DisablePayloadSigning = true
                };

                // initialise the transfer/upload tools
                var transferUtility = new TransferUtility(client);

                // initiate the file upload
                await transferUtility.UploadAsync(uploadRequest);

                response.StatusCode = 201;
                response.Message = $"{obj.Name} has been uploaded successfully";
            }
            catch (AmazonS3Exception s3Ex)
            {
                response.StatusCode = (int)s3Ex.StatusCode;
                response.Message = s3Ex.Message;
            }
            catch (Exception ex)
            {
                response.StatusCode = 500;
                response.Message = ex.Message;
            }
            return response;

The expected behaviour is a successful upload of the file. 
The parenthesis is on the safe character list. 
Uploading the file only with underscores _ is working!
Do you have any ideas how to get this problem solved? 

Best Regards,
Fabian

**Comment by dscpinheiro** [comment]
Sorry for the delay in the response, but what version of S3 are you using? Also, what does `settings.ObsAccessPoint` refer to in your environment?

Using `AWSSDK.S3 3.7.105.1`, I was able to upload that file name to my access point:
```
var config = new AmazonS3Config
{
    ForcePathStyle = true
};

var client = new AmazonS3Client(config);
var uploadRequest = new TransferUtilityUploadRequest
{
    FilePath = "C:\\Temp\\file_(test)_file.png",
    Key = "file_(test)_file.png",
    BucketName = "myap-random-string-s3alias",
    CannedACL = S3CannedACL.NoACL,
};

var transferUtility = new TransferUtility(client);
await transferUtility.UploadAsync(uploadRequest);
```
